### PR TITLE
Fold MockWebServerRule into MockWebServer.

### PR DESCRIPTION
**Pull request by JakeWharton**
This avoid needing to duplicate the methods on the rule which historically was a subset of everything.

JUnit is now a required dependency.

### CODE REVIEW

**Comment by swankjesse** [comment]
Would it be simpler to flatten the hierarchy?

**Comment by swankjesse** [comment]
LGTM

**Comment by JakeWharton** [comment]
There was a reason I didn't about 6 months ago. I'll try it and see if I
can rediscover why.

**Comment by JakeWharton** [comment]
PTAL!

File: mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java [hunk 1]
```diff
@@ -127,44 +129,61 @@
...
   @Deprecated
   public URL getUrl(String path) {
-    try {
-      return sslSocketFactory != null
-          ? new URL("https://" + getHostName() + ":" + getPort() + path)
-          : new URL("http://" + getHostName() + ":" + getPort() + path);
-    } catch (MalformedURLException e) {
-      throw new AssertionError(e);
-    }
+    return url(path).url();
```
Comment (JakeWharton) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java]: Winning

File: mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java [hunk 2]
```diff
@@ -177,8 +196,8 @@ public HttpUrl url(String path) {
         .scheme(sslSocketFactory != null ? "https" : "http")
         .host(getHostName())
         .port(getPort())
-        .encodedPath(path)
-        .build();
+        .build()
+        .resolve(path);
```
Comment (JakeWharton) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java]: This allows values like `/?foo=bar` to work.

Comment (swankjesse) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java]: Also winning

**Pull request merged**